include/boost/geometry/algorithms/assign.hpp

```diff
@@ -236,20 +236,16 @@ struct assign
             
         static bool const same_point_order
             = point_order<Geometry1>::value == point_order<Geometry2>::value;
-        BOOST_MPL_ASSERT_MSG
-        (
-            (same_point_order),
-            ASSIGN_IS_NOT_SUPPORTED_FOR_DIFFERENT_POINT_ORDER,
-            (types<Geometry1, Geometry2>)
-        );
+        BOOST_GEOMETRY_STATIC_ASSERT(
+            same_point_order,
+            "Assign is not supported for different point orders.",
+            Geometry1, Geometry2);
```

Comment:
This style looks really good to me.

Reply:
Yes, this is possible thanks to variable-argument macros added in C++11.

Btw, have in mind that `static_assert` is slightly different than `BOOST_MPL_ASSERT`. A condition used in it has to be dependent on a template parameter. Otherwise the `static_assert` will always fail or pass even if a template is not instantiated. E.g. a program like this:
```
template <typename T>
struct A
{
    static_assert(false);
};

int main()
{}
```
will always fail to compile. This is why I also added `BOOST_GEOMETRY_STATIC_ASSERT_FALSE` which requires dependent types and deals with this problem.